docs/architecture/views.md

```diff
@@ -0,0 +1,169 @@
+      onclick: FindThatDisk,
+    },
+    text("It's in that place where I put that thing that time.")
+  )
+```
+<!-- In the 1995 movie "Hackers", the hacker "The Phantom Freak" calls his friend "Acid Burn" from jail as he's being framed for a crime he didn't commit. -->
+
+[`text()`](../api/text.md) just creates text nodes so the views it can create on its own are necessarily simplistic.
+
+```js
+const view = () => text("Go home and be a family man!")
+```
+<!-- In the videogame "Street Fighter II: The World Warrior", the fighter known as Guile says this taunt to his opponent after defeating them. -->
+
+[`memo()`](../api/memo.md) is designed to be used with other functions that produce VNodes, so it doesn't really describe a view by itself.
+
+```js
+const view = (state) => memo(scenicView, state.vacationSpot)
+```
+<!-- Just a play-on-words between how "view" is used in Hyperapp and everyday language. -->
+
+---
+
+## Components
+
+Views are naturally composable so they can be as simple or complicated as you need. Simpler apps probably just need a single view but in more complicated apps there could be plenty of subviews.
+
+A **component** in Hyperapp can either be a subview or some other function that generates VNodes.
+
+You would typically make components for widgets that provide the building block elements of your app's UI. Components for larger UI segments such as dashboards or pages would make use of these widgets.
+
+In the following example, `coinsDisplay` is a component in the form of a subview while `questionBlock` is a component in the form of some function that returns a VNode. Notice the former cares directly about the state while the latter doesn't:
+
+```js
+const coinsDisplay = (state) =>
+  h("div", { class: "coins-display" }, text(state.coins))
+
+const questionBlock = (opened) =>
+  opened
+    ? h("button", { class: "question-block opened" }, text("?"))
+    : h(
+        "button",
+        {
+          class: "question-block",
+          onclick: [
+            HitBlockFromBottom,
+            { revealItem: "beanstalk" },
+          ],
+        },
+        text("?")
+      )
+
+const level = (state) =>
+  h("div", { class: "level" }, [
+    coinsDisplay(state),
+    questionBlock(state.onlyQuestionBlockOpened),
+  ])
+```
+<!-- In the videogame "Super Mario Bros." coins are important for earning extra lives and the question blocks often contain useful contents. -->
+
+### Components Returning Multiple VNodes
+
+Components are allowed to return an array of VNodes. However, to make use of such components you'll need to spread their result.
+
+```js
+const finishingMoveOptions = () => [
+  h("button", { onclick: FinishHim }, text("Fatality")),
+  h("button", { onclick: FinishHimAsAnAnimal }, text("Animality")),
+  h("button", { onclick: TurnHimIntoABaby }, text("Babality")),
+  h("button", { onclick: BefriendHim }, text("Friendship")),
+]
+
+const view = () => h("div", {}, ...finishingMoveOptions())
```

Comment:
This example is incorrect. `h` cannot accept arbitrarily many arguments any longer. I think you meant to say either:

```js
const view = () => h("div", {}, finishingMoveOptions())
```

or 

```js
const view = () => h("div", {}, [...finishingMoveOptions()])
```

Reply:
Nice catch! I'll go with the first option.